File: Day2/code.java
import java.io.IOException;
import java.nio.file.Path;
import java.util.Scanner;

public class code{
    public static void main(String[] args) throws IOException {
        Scanner file = new Scanner(Path.of("input.txt"));
        System.out.printf("The answer to part 1 is:\n%d\n",part1(file));
        file = new Scanner(Path.of("input.txt"));
        System.out.printf("The answer to part 2 is:\n%d\n",part2(file));
        file.close();
    }

    static int part1 (Scanner file) {
        return 0;
    }
    static int part2 (Scanner file) {
        return 0;
    }
}

File: Day2/code.py
def main():
    with open("input.txt","r") as f:
        print(f"Answer for Part 1:\n{part1(f)}") 
    with open("input.txt","r") as f:
        print(f"Answer for Part 2:\n{part2(f)}") 

def part1(f:list[str]):
    total = 0
    for line in f:
        ranges = line.split(",")
    ranges = [range_bounds.split("-") for range_bounds in ranges]
    for range_bounds in ranges:
        for number in range(int(range_bounds[0]),int(range_bounds[1])+1):
            number = str(number)
            if number[:int(len(number)/2)] == number[int(len(number)/2):]:
                total += int(number)
    return total

def part2(f):
    total = 0
    for line in f:
        ranges = line.split(",")
    ranges = [range_bounds.split("-") for range_bounds in ranges]
    for range_bounds in ranges:
        for number in range(int(range_bounds[0]),int(range_bounds[1])+1):
            number = str(number)
            midpoint = int(len(number)/2)
            for cutoff in range(midpoint,0,-1):
                if number[cutoff:] == number[:cutoff]:
                    total += int(number)
                    break
    return total

main()